batch_test.go

```diff
@@ -0,0 +1,47 @@
+package ed25519consensus
+
+import (
+	"crypto/ed25519"
+	"fmt"
+	"testing"
+)
+
+func TestBatch(t *testing.T) {
```

Comment:
In addition to testing that a batch of all valid signatures passes, it'd be great to test that having a batch with an invalid element fails.  One way to do this would be to create a valid batch, and then make it invalid in different ways:

* changing the message of one of the batch items;
* modifying a byte (say, the low byte) of one of the batch items' `R` values;
* modifying a byte (say, the low byte) of one of the batch items' `s` values.

Each of these modifications should hit a different code path, so it probably makes sense to have them as three different invalid batch tests.